bin/qgis/smoderp2d-plugin/smoderp_2D_dockwidget.py

```diff
@@ -473,7 +473,8 @@ def _getInputParams(self):
         # TODO: It would be nicer to use names defined in _input_params before
         # this reparsing
         for key in self._input_maps.keys():
-            self._input_params[key] = key
+            if self._input_params[key]:
```

Comment:
What is this supposed to do? Is it supposed to check if the value of `self._input_params[key]` is not `None`?

Reply:
Here are dict values replaced by keys, so 

```
self._input_params['streams'] = 'elevation'
...
self._input_params['streams'] = 'streams'
```

Later the values are used for map names in GRASS location. Unfortunately 

```
self._input_params['streams'] = 'streams'
```

leads to the error reported in #252.

Reply:
And back to your question. We check `None` value and empty string.

Reply:
> And back to your question. We check None value and empty string.

This is what I was for. Why do we check for `None`? The value should never be `None`, but a string instead. If there is `None`, something went wrong. Wouldn't it be better to replace it with a specific one, `if self._input_params[key] == '':`?

(PS: The current clause checks also for `0`)